site/source/docs/porting/pthreads.rst

```diff
@@ -4,50 +4,58 @@
+Additional flags
+================
 
-There should be no other changes required. In C/C++ code, the preprocessor check #ifdef __EMSCRIPTEN_PTHREADS__ can be used to detect whether Emscripten is currently targeting pthreads.
+- ``-s PROXY_TO_PTHREAD``: In this mode your original ``main()`` is replaced by
+  a new one that creates a pthread and runs the original ``main()`` on it. As a
+  result, your application's main code is run off the main thread, which is good
+  for responsiveness. The main thread does still run code when things are
+  proxied to it, for example to handle events, rendering, etc.
+
+Note that Emscripten has ``--proxy-to-worker`` which sounds similar but is
+unrelated. That flag does not use pthreads or SharedArrayBuffer, and instead
+uses a plain Web Worker to run your main program (and postMessage to proxy
+messages back and forth).
 
 Special considerations
 ======================
 
 The Emscripten implementation for the pthreads API should follow the POSIX standard closely, but some behavioral differences do exist:
 
-- At runtime, you can `#include <emscripten/threading.h>` and use the ``emscripten_has_threading_support()`` function to
-  test whether the currently executing code was compiled with pthreads support
-  enabled. If this function returns true, then the currently executing code was
-  compiled with ``-s USE_PTHREADS=1`` (and the current browser supports
-  multithreading).
-
-  If code is compiled with ``-s USE_PTHREADS=1`` and the current browser does
-  not support multithreading, then an exception will be thrown at page load
-  time.  It is not possible to build one binary that would be able to leverage
-  multithreading when available and fall back to single threaded when not. For
-  such backwards compatibility, two separate builds must be done, one with ``-s
-  USE_PTHREADS=1`` and the other with ``-s USE_PTHREADS=0``.
-
-- When the linker flag `-s PTHREAD_POOL_SIZE=<integer>` is not specified and `pthread_create()` is called, the new thread will not actually start to run immediately, but the main JS thread must yield execution back to browser first. This behavior is a result of `#1049079 <https://bugzilla.mozilla.org/show_bug.cgi?id=1049079>`_.
+- When the linker flag ``-s PTHREAD_POOL_SIZE=<integer>`` is not specified and ``pthread_create()`` is called, the new thread will not actually start to run immediately, but you must yield execution back to the browser's main event loop first (in order for ``new Worker`` and the related events to happen). If you don't wait but synchronously expect it to be ready (like if you join on it, or if you look at memory where you expect it to write) things may not work. Using a pool allows things to work more like a native build.
```

Comment:
It might be clearer to say something like "the new thread will not start until control is yielded back to the browser's main event loop, because the web worker cannot be created while JS or wasm code is running. This is a violation of POSIX behavior and will break common code which creates a thread and immediately joins it or otherwise waits to observe an effect such as a memory write. Using a pool creates the web workers before `main` is called, allowing thread creation to be synchronous."